lib/compiler-singlepass/src/lib.rs

```diff
@@ -19,3 +19,6 @@ mod x64_decl;
 
 pub use crate::compiler::SinglepassCompiler;
 pub use crate::config::Singlepass;
+
+#[macro_use]
+extern crate memoffset;
```

Comment:
It's more idiomatic to not write `extern crate` at all, and just do a normal

```
use memoffset::offset_of;
```

Reply:
That's how they recommended it crate docs.

Reply:
And it doesn't seem to work in suggested way.

Reply:
Yeah, the docs are just old, `extern crate` is no longer necessary since Rust 2018: https://doc.rust-lang.org/edition-guide/rust-2018/path-changes.html#no-more-extern-crate

The following diff works for me:

```diff
diff --git a/lib/compiler-singlepass/src/codegen_x64.rs b/lib/compiler-singlepass/src/codegen_x64.rs
index 41695196a..64500ca5a 100644
--- a/lib/compiler-singlepass/src/codegen_x64.rs
+++ b/lib/compiler-singlepass/src/codegen_x64.rs
@@ -2,6 +2,7 @@ use crate::address_map::get_function_address_map;
 use crate::config::{Intrinsic, IntrinsicKind};
 use crate::{common_decl::*, config::Singlepass, emitter_x64::*, machine::Machine, x64_decl::*};
 use dynasmrt::{x64::Assembler, DynamicLabel};
+use memoffset::offset_of;
 use smallvec::{smallvec, SmallVec};
 use std::collections::HashMap;
 use std::iter;
diff --git a/lib/compiler-singlepass/src/lib.rs b/lib/compiler-singlepass/src/lib.rs
index 2afb629db..f0dada26e 100644
--- a/lib/compiler-singlepass/src/lib.rs
+++ b/lib/compiler-singlepass/src/lib.rs
@@ -20,5 +20,5 @@ mod x64_decl;
 pub use crate::compiler::SinglepassCompiler;
 pub use crate::config::Singlepass;
 
-#[macro_use]
-extern crate memoffset;
+// #[macro_use]
+// extern crate memoffset;
```